src/data_collector/polygon_data/client.py

```
@@ -136,27 +152,36 @@ def _make_request(self, endpoint: str, params: Optional[Dict] = None) -> Dict[st
+                if getattr(config, 'DISABLE_RATE_LIMITING', False):
+                    logger.warning("Connection error. Retrying without delay (rate limiting disabled)")
+                else:
+                    logger.warning(f"Connection error. Retrying in {wait_time}s")
+                    if attempt < config.MAX_RETRIES - 1:
+                        time.sleep(wait_time)
+                        continue
                 raise PolygonAPIError("Connection error after all retries")
             
             except requests.exceptions.RequestException as e:
                 if attempt == config.MAX_RETRIES - 1:
                     raise PolygonAPIError(f"Request failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)